### ISSUE TITLE
[Bug?] Ships push custom pads downward [20w23c]

### ISSUE THREAD

**Issue by Dwarf-LordPangolin** [tsunamayo/Starship-EVO#2335]
### Short Description of the issue:
Wasn't sure this was a bug or just a functionality that was not implemented yet, but just in case: if attempting to land on a pad with a shipyard block on it, the pad will not hold still, but will be continuously pushed down by the landing ship.

### Steps to reproduce the issue:
Create custom landing pad and try to land on it.

### Environment
- Starship EVO version: 20w23c
- Graphics Card: NVIDIA GeForce GTX 1060 6GB

### Additional information:
[output_log.txt](https://github.com/tsunamayo/Starship-EVO/files/4728354/output_log.txt)

**Comment by tsunamayo** [comment]
Yes it is still wip, I need to desactivate the physics on the shipyard. My only problem was again the UX side, ie what about user created stations?
My best plan was an anchor block of some sort...
I think I will also need a broadcast antenna, to say how far your station is visible.

**Comment by tsunamayo** [comment]
btw latest pcg is now up / down aware.

![image](https://user-images.githubusercontent.com/6604318/83731217-1c11a100-a685-11ea-97d3-16d09884bec1.png)

I am gonna add more rule, ie for example put living module on top, landingpad on the exterior middle, and maintenance modules on the bottom. If you have more idea please shoot!

**Comment by Dwarf-LordPangolin** [comment]
Wow, the procedural generation looks like it's working great! Would you like me to make the vertical parts of the 6-way connectors symmetrical, so they line up more cleanly?

Those rules make sense, especially putting the pads on the exterior. I can't think of any more ideas, but I'll ask the others if they have any.

**Comment by ProPeach** [comment]
Your system looks awesome so far! Really looking forward to docking my ship at one :)
Talking a bit more about layout on the Discord, we were wondering if a bit more variation in density would be possible - some areas like residential would look nice being more tightly packed, while some areas should be more spread out.
Also in regards to orientation and general footprint, would it be possible to favour a more vertical layout or horizontal layout and try to stray away from the more spherical "blob" shaped stations? There should be examples of all profiles thanks to the stations being randomly generated of course, but I think favouring one dimension over the other can make for more consistently interesting layouts rather than one central grouping most of the time. Overall shape of the station will be most important in making each one look unique and fascinating in my opinion, some truly wacky designs would be awesome!

Other than that as players build more modules like observatories, communications arrays or other "flavour" modules, these should also generate on the edges of stations. Perhaps these more interesting modules need their own category to ensure that?
I can't wait to see how this progresses!

**Comment by tsunamayo** [comment]
Yes @dwarf maybe you could tweak it a bit to break it, but thats not a big deal, I would say if you do something do other modules! I will put up a list soon, I can start to see the fun in this.
@ProPeach yes exactly, I will add some variables driving the generation, the verticality being one of the factor. That is why I added the Up / Down logic, so I can control this. I will add plenty of parameters to your modules to be used in the PCG algo.
We will add a lot of other modules and category! Turrets modules, tanks clusters, med bays, trading outpost, police stations and what not, anything you can think of. We will iterate quite a bit, I will try to add this little by little.
For example some stations will have themes, some being more trading hub, living outpost, entertainment hub, ect ect. This will dictate the probability of spawning different modules type.
